src/EA.Weee.Web/Areas/AATFReturn/Controllers/SelectYourPCSController.cs

```diff
@@ -20,30 +22,58 @@ public class SelectYourPCSController : ExternalSiteController
     {
         private readonly Func<IWeeeClient> apiClient;
         private readonly BreadcrumbService breadcrumb;
+        private readonly IAddReturnSchemeRequestCreator requestCreator;
         private readonly IWeeeCache cache;
-        private List<SchemeData> schemeList;
 
-        public SelectYourPCSController(Func<IWeeeClient> apiclient, BreadcrumbService breadcrumb, IWeeeCache cache)
+        public SelectYourPCSController(Func<IWeeeClient> apiclient, BreadcrumbService breadcrumb, IWeeeCache cache, IAddReturnSchemeRequestCreator requestCreator)
         {
             this.apiClient = apiclient;
             this.breadcrumb = breadcrumb;
             this.cache = cache;
+            this.requestCreator = requestCreator;
         }
 
         [HttpGet]
-        public virtual async Task<ActionResult> Index(Guid organisationId)
+        public virtual async Task<ActionResult> Index(Guid organisationId, Guid returnId)
         {
-            var viewModel = new SelectYourPCSViewModel();
+            var viewModel = new SelectYourPCSViewModel()
+            {
+                OrganisationId = organisationId,
+                ReturnId = returnId
+            };
 
             using (var client = apiClient())
             {
-                var schemeRequest = new GetSchemes();
-                schemeList = await client.SendAsync(User.GetAccessToken(), schemeRequest);
-                viewModel.SchemeList = schemeList;
+                viewModel.SchemeList = await client.SendAsync(User.GetAccessToken(), new GetSchemesExternal());
             }
             await SetBreadcrumb(organisationId, BreadCrumbConstant.AatfReturn);
 
-            return View(viewModel);
+            return View("Index", viewModel);
+        }
+
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public virtual async Task<ActionResult> Index(SelectYourPCSViewModel viewModel)
+        {
+                if (ModelState.IsValid)
+            {
+                using (var client = apiClient())
+                {
+                    var requests = requestCreator.ViewModelToRequest(viewModel);
+
+                    foreach (var request in requests)
+                    {
```

Comment:
Think Sam is working on this, don't make multiple requests to the api create a request that contains the list and do single request